### PR TITLE
Add `set` and `add` TensorVariable methods for `set_subtensor` and `inc_subtensor` operations`

### PR DESCRIPTION
**Pull request by ricardoV94**
Changes #493 

New API:
```python
import pytensor.tensor as pt

x = pt.matrix()
y = x.set(0, 5)  # Equivalent to `y = pt.set_subtensor(x[0], 5)`
y = x.add(0, 5)  # Equivalent to `y = pt.inc_subtensor(x[0], 5)`
```

### CODE REVIEW

File: pytensor/tensor/variable.py [hunk 1]
```diff
@@ -816,7 +816,9 @@ def compress(self, a, axis=None):
         return at.extra_ops.compress(self, a, axis=axis)
 
     def set(self, y, **kwargs):
-        """Set values to y, where y is the output of an index operation.
+        """Return a tensor whose indexed values are set to y.
```
Comment (lucianopaz) [inline pytensor/tensor/variable.py:819]:
```suggestion
        """Return a copy of the calling tensor, where the indexed values are set to y.
```

File: pytensor/tensor/variable.py [hunk 2]
```diff
@@ -829,9 +831,11 @@ def set(self, y, **kwargs):
         return at.subtensor.set_subtensor(self, y, **kwargs)
 
     def add(self, y, **kwargs):
-        """Add values to y, where y is the output of an index operation.
+        """Return a tensor whose indexed values are incremented by y.
```
Comment (lucianopaz) [inline pytensor/tensor/variable.py:834]:
```suggestion
        """Return a copy of the calling tensor where the indexed values are incremented by y.
```

**Review by lucianopaz — CHANGES_REQUESTED** [review]
I think that we should explicitly say that the results are copies of the original tensor and not references to it.

**Comment by ricardoV94** [comment]
Would the API be more intuitive if you passed the indexes as well?

```python
x = x[idx].set(y)
x = x.set(idx, y)
```

File: pytensor/tensor/variable.py (hunk 1 above)
Comment (lucianopaz) [reply pytensor/tensor/variable.py]: I meant self. I’m not sure how to call it

Comment (ricardoV94) [reply pytensor/tensor/variable.py]: But, it's not self, because this is not a `__getitem__`. self is the subtensor operation `x[idx]`.

**Comment by lucianopaz** [comment]
> Would the API be more intuitive if you passed the indexes as well?
> 
> ```python
> x = x[idx].set(y)
> x = x.set(idx, y)
> ```

Yes, it also follows the order of arguments that `__setitem__` has

**Review by lucianopaz — APPROVED** [review]

**Pull request merged**